src/AppInstallerCommonCore/Versions.cpp

```diff
@@ -275,4 +355,77 @@ namespace AppInstaller::Utility
 
         return m_version < other.m_version;
     }
+
+    VersionRange::VersionRange(Version minVersion, Version maxVersion)
+    {
+        THROW_HR_IF(E_INVALIDARG, minVersion > maxVersion);
+        m_minVersion = std::move(minVersion);
+        m_maxVersion = std::move(maxVersion);
+    }
+
+    bool VersionRange::HasOverlapWith(const VersionRange& other) const
+    {
+        // No overlap if either is an empty range.
+        if (IsEmpty() || other.IsEmpty())
+        {
+            return false;
+        }
+
+        return m_minVersion <= other.m_maxVersion && m_maxVersion >= other.m_minVersion;
+    }
+
+    bool VersionRange::IsSameAsSingleVersion(const Version& version) const

```

Comment:
It seems like this could be an operator overload of `==` instead of a separate method?

Reply:
I don't feel like doing `==` for different objects, which may cause confusion IMO.